enterprise/server/scim/scim.go

```diff
@@ -29,68 +99,236 @@ func NewSCIMServer(env environment.Env) *SCIMServer {
 	}
 }
 
-func (s *SCIMServer) Users() http.Handler {
-	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+type handlerFunc func(ctx context.Context, r *http.Request, g *tables.Group) (interface{}, error)
+
+func mapErrorCode(err error) int {
+	if status.IsNotFoundError(err) {
+		return http.StatusNotFound
+	} else if status.IsInvalidArgumentError(err) {
+		return http.StatusBadRequest
+	}
+	return http.StatusInternalServerError
+}
+
+func (s *SCIMServer) RegisterHandlers(mux interfaces.HttpServeMux) {
+	fn := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if !strings.HasPrefix(r.URL.Path, usersPath) {
+			w.WriteHeader(http.StatusNotFound)
+			return
+		}
+
+		u, err := s.env.GetAuthenticator().AuthenticatedUser(r.Context())
+		if err != nil {
+			w.WriteHeader(http.StatusForbidden)
+			return
+		}
+		if !u.HasCapability(akpb.ApiKey_SCIM_CAPABILITY) {
+			w.WriteHeader(http.StatusForbidden)
+			return
+		}
+		g, err := s.env.GetUserDB().GetGroupByID(r.Context(), u.GetGroupID())
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			w.Write([]byte("could not lookup group information"))
+			return
+		}
+		if g.SamlIdpMetadataUrl == nil || *g.SamlIdpMetadataUrl == "" {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte("SCIM API can only be used in conjunction with SAML"))
+			return
+		}
+
+		var h handlerFunc
 		switch r.Method {
 		case "GET":
-			s.getUser(w, r)
+			if r.URL.Path == usersPath {
```

Comment:
Is your intention to remove the group path altogether? Or is that just coming later?

Reply:
it will be back, has to be redone anyway